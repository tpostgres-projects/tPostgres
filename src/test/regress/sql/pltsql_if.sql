--
-- PLTSQL -- IF Statement
--

CREATE FUNCTION if_with_semicolon() RETURNS void AS $$
BEGIN
    IF true THEN
        PRINT 'true'
    END IF;
END
$$ LANGUAGE pltsql;

SELECT if_with_semicolon();

CREATE FUNCTION if_without_semicolon() RETURNS void AS $$
BEGIN
    IF true THEN
        PRINT 'true'
    END IF
END
$$ LANGUAGE pltsql;

SELECT if_without_semicolon();

CREATE FUNCTION if_without_then() RETURNS void AS $$
BEGIN
    IF true
        PRINT 'true'
    END IF
END
$$ LANGUAGE pltsql;

SELECT if_without_then();

CREATE FUNCTION if_elsif_without_then() RETURNS void AS $$
DECLARE @a int
BEGIN
    SET @a = 1
    IF 2 > @a
        UPDATE pg_settings SET setting = 'notupdated' WHERE name = 'nonexistent';
    ELSIF @a = 1
        SET @a = 2
    ELSIF @a = 2
        SET @a = 3
    ELSE
        SET @a = NULL
    END IF
END
$$ LANGUAGE pltsql;

SELECT if_elsif_without_then();
