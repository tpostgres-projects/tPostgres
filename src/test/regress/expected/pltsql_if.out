--
-- PLTSQL -- IF Statement
--
CREATE FUNCTION if_with_semicolon() RETURNS void AS $$
BEGIN
    IF true THEN
        PRINT 'true'
    END IF;
END
$$ LANGUAGE pltsql;
SELECT if_with_semicolon();
INFO:  true
 if_with_semicolon 
-------------------
 
(1 row)

CREATE FUNCTION if_without_semicolon() RETURNS void AS $$
BEGIN
    IF true THEN
        PRINT 'true'
    END IF
END
$$ LANGUAGE pltsql;
SELECT if_without_semicolon();
INFO:  true
 if_without_semicolon 
----------------------
 
(1 row)

CREATE FUNCTION if_without_then() RETURNS void AS $$
BEGIN
    IF true
        PRINT 'true'
    END IF
END
$$ LANGUAGE pltsql;
SELECT if_without_then();
INFO:  true
 if_without_then 
-----------------
 
(1 row)

CREATE FUNCTION if_elsif_without_then() RETURNS void AS $$
DECLARE @a int
BEGIN
    SET @a = 1
    IF 2 > @a
        UPDATE pg_settings SET setting = 'notupdated' WHERE name = 'nonexistent';
    ELSIF @a = 1
        SET @a = 2
    ELSIF @a = 2
        SET @a = 3
    ELSE
        SET @a = NULL
    END IF
END
$$ LANGUAGE pltsql;
SELECT if_elsif_without_then();
 if_elsif_without_then 
-----------------------
 
(1 row)

